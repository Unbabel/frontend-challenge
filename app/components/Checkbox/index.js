import React, { useState, useContext } from 'react';
import { ThemeContext } from 'styled-components';

import Label from './Label';
import Input from './Input';
import Span from './Span';

function Checkbox() {
  const [isChecked, setChecked] = useState(false);

  return (
    <Label theme={useContext(ThemeContext)}>
      <Input checked={isChecked} onChange={() => setChecked(!isChecked)} type="checkbox" />
      <Span>
        <span />
      </Span>
    </Label>
  );
}

export default Checkbox;
