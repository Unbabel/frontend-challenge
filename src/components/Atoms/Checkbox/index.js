import React, { useState } from 'react'
import styled from 'styled-components'
import { prop } from 'styled-tools'


const CheckboxContainer = styled.div`
  display: inline-block;
  vertical-align: middle;
  margin: .25rem;
`

const Icon = styled.svg`
  fill: none;
  stroke: white;
  stroke-width: 2px;
`
// Hide checkbox visually but remain accessible to screen readers.
// Source: https://polished.js.org/docs/#hidevisually
const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  border: 0;
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
  
`

const StyledCheckbox = styled.div`
  display: inline-block;
  width: 16px;
  height: 16px;
  background: ${props => (props.checked ? '#ffffff' : '#859EFF')}
  border-radius: 3px;
  border: 3px solid #859EFF;
  transition: all 150ms;
  cursor:pointer;
  line-height:1rem;

  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px pink;
  }

  ${Icon} {
    visibility: ${props => (props.checked ? 'hidden' : 'visible')}
  }
`

const Checkbox = ({ ...props }) => {
  const [checkedItem, setCheckedItem] = useState({ value: false })

  const handleChange = (event) => {
    setCheckedItem(!checkedItem)
    console.log('checkedItem: ', checkedItem)
  }
  return (
    <CheckboxContainer>
      <HiddenCheckbox {...props} />
      <StyledCheckbox {...props} checked={checkedItem} onClick={handleChange}>
        <Icon viewBox="0 0 24 24">
          <polyline points="20 6 9 17 4 12" />
        </Icon>
      </StyledCheckbox>
    </CheckboxContainer>
  )
}
export default Checkbox
