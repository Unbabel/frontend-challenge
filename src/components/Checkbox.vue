<template>
    <label class="custom-checkbox">
        <input name="custom checkbox" type="checkbox">
        <span class="checkmark"></span>
    </label>
</template>

<script>
export default {
    name: 'Checkbox'
}
</script>

<style lang="scss" scoped>
    .custom-checkbox {
        position: relative;
        border: 1px solid #859EFF;
        width: 16px;
        height: 16px;
        border-radius: 1px;
        &:hover {
            cursor: pointer;
        }
        input {
            opacity: 0;
            &:checked ~ .checkmark {
                background-color: #859EFF;
            }
            &:checked ~ .checkmark:after {
                opacity: 1;
            }
        }
        .checkmark {
            display: block;

            position: absolute;
            top: 0;
            left: 0;
            height: 15px;
            width: 15px;
            background-color: transparent;
            &:after {
                content: "";
                border: 2px solid #fff;
                border-left: 0;
                border-top: 0;
                height: 9px;
                width: 4px;
                left: 4px;
                top: 0;
                border-radius: 1px;
                opacity: 0;
                position: absolute;
                transform: rotate(45deg);
                transition: opacity 0.2s ease-in-out;
            }
        }
    }
</style>


