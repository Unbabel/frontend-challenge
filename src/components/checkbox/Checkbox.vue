<script setup>
	import { computed } from "vue";
	const { id } = defineProps({
		id: {
			type: String,
			required: true,
		},
	});

	const checkboxId = computed(() => "checkbox-" + id);
</script>

<template>
	<div class="checkbox-div">
		<input
			name="cssCheckbox"
			:id="checkboxId"
			type="checkbox"
			class="css-checkbox"
			aria-checked
			checked
		/>
		<label :for="checkboxId"></label>
	</div>
</template>

<style scoped>
	.checkbox-div {
		height: 1rem;
		width: 1rem;
	}

	.css-checkbox {
		position: absolute;
		overflow: hidden;
		clip: rect(0 0 0 0);
		height: 0.0625rem;
		width: 0.0625rem;
		padding: 0;
		border: 0;
	}

	.css-checkbox + label {
		position: relative;
		font-size: 1rem;
		cursor: pointer;
		display: inline-flex;
		align-items: center;
	}

	.css-checkbox + label::before {
		content: " ";
		vertical-align: middle;
		height: 0.75rem;
		width: 0.75rem;
		border-width: 0.125rem;
		border-style: solid;
		border-color: var(--palette-purple);
		border-radius: 0.125rem;
		box-shadow: none;
	}

	.css-checkbox:checked + label::after {
		content: " ";
		background-image: url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA1MTIgNTEyIj48cGF0aCBmaWxsPSIjZmZmZmZmIiBkPSJNMTczLjg5OCA0MzkuNDA0bC0xNjYuNC0xNjYuNGMtOS45OTctOS45OTctOS45OTctMjYuMjA2IDAtMzYuMjA0bDM2LjIwMy0zNi4yMDRjOS45OTctOS45OTggMjYuMjA3LTkuOTk4IDM2LjIwNCAwTDE5MiAzMTIuNjkgNDMyLjA5NSA3Mi41OTZjOS45OTctOS45OTcgMjYuMjA3LTkuOTk3IDM2LjIwNCAwbDM2LjIwMyAzNi4yMDRjOS45OTcgOS45OTcgOS45OTcgMjYuMjA2IDAgMzYuMjA0bC0yOTQuNCAyOTQuNDAxYy05Ljk5OCA5Ljk5Ny0yNi4yMDcgOS45OTctMzYuMjA0LS4wMDF6Ii8+PC9zdmc+");
		background-repeat: no-repeat;
		background-size: 0.6rem 1rem;
		background-position: center center;
		position: absolute;
		display: flex;
		justify-content: center;
		align-items: center;
		text-align: center;
		background-color: var(--palette-purple);
		font-size: 1rem;
		height: 0.875rem;
		width: 0.985rem;
		border-radius: 0.125rem;
	}
</style>
