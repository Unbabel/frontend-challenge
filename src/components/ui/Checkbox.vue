<template>
  <div>
    <input type="checkbox" :id="id">
    <label :for="id"></label>
  </div>
</template>

<script>

export default {
  name: 'Checkbox',
  props: {
    id: Number
  }
}
</script>

<style scoped lang="scss">
@import '@/scss/main.scss';

input {
  display: none;
}

label {
  width: 16px;
  height: 16px;
  border-radius: 2px;
  border: 2px solid $checkbox-border-color;
  flex-shrink: 0;
  cursor: pointer;
  position: relative;
  margin: 5px 16px 0 0;
  display: block;

  &::before {
    content: '';
    position: absolute;
    top: 2px;
    left: 1px;
    width: 11px;
    height: 8px;
    z-index: 1;
    background: url('~@/assets/images/check-mark.svg') center / cover no-repeat;
    display: none;
  }
}

input:checked + label {
  background-color: $checkbox-border-color;

  &::before {
    display: block;
  }
}
</style>
