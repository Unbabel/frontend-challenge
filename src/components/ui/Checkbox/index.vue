<template>
  <label :for="`checkbox-${id}`">
    <input type="checkbox" :id="`checkbox-${id}`" />
    <span class="box">
      <span class="arrow"></span>
    </span>
  </label>
</template>
<script>
export default {
  name: "Checkbox",

  props: {
    id: {
      required: true,
      type: Number
    }
  }
};
</script>
<style scoped lang="scss">
label {
  display: inline-block;
  position: relative;
}

input {
  position: absolute;
  left: -99999px;
}

.box {
  display: inline-flex;
  cursor: pointer;
  min-height: 16px;
  padding-left: calc(16px + 0.4em);
}

.box:before,
.box:after {
  content: "";
  box-sizing: border-box;
  width: 1em;
  height: 1em;
  font-size: 16px;

  position: absolute;
  left: 0;
  top: 0;
}

.box:before {
  border: 2px solid $color-purple;
  z-index: 2;
}

input:disabled ~ .box:before {
  border-color: $color-grey;
}

input:not(:disabled) ~ .box:after {
  background-color: $color-purple;
  opacity: 0;
}

input:not(:disabled):checked ~ .box:after {
  opacity: 1;
}

.arrow {
  margin-top: auto;
  margin-bottom: auto;
}

/*
  The arrow size and position depends from sizes of square because I needed an arrow correct positioning from the top left corner of the element toggle
  */

.arrow:before {
  content: "";
  box-sizing: border-box;
  width: 0;
  height: 0;
  font-size: 16px;

  border-left-width: 0;
  border-bottom-width: 0;
  border-left-style: solid;
  border-bottom-style: solid;
  border-color: $color-white;

  position: absolute;
  top: 0.5428em;
  left: 0.2em;
  z-index: 3;

  transform-origin: left top;
  transform: rotate(-40deg) skew(10deg);
}

input:not(:disabled):checked ~ .box .arrow:before {
  width: 0.5em;
  height: 0.25em;
  border-left-width: 2px;
  border-bottom-width: 2px;
  will-change: width, height;
  transition: width 0.1s ease-out 0.2s, height 0.2s ease-out;
}

/*
  =====
  LEVEL 2. PRESENTATION STYLES
  =====
  */

/*
  The demo skin
  */

.box:before,
.box:after {
  border-radius: 2px;
}

/*
  The animation of switching states
  */

input:not(:disabled) ~ .box:before,
input:not(:disabled) ~ .box:after {
  opacity: 1;
  transform-origin: center center;
  will-change: transform;
  transition: transform 0.2s ease-out;
}

input:not(:disabled) ~ .box:before {
  transform: rotateY(0deg);
  transition-delay: 0.2s;
}

input:not(:disabled) ~ .box:after {
  transform: rotateY(90deg);
}

input:not(:disabled):checked ~ .box:before {
  transform: rotateY(-90deg);
  transition-delay: 0s;
}

input:not(:disabled):checked ~ .box:after {
  transform: rotateY(0deg);
  transition-delay: 0.2s;
}

.arrow:before {
  opacity: 0;
}

input:not(:disabled):checked ~ .box .arrow:before {
  opacity: 1;
  transition: opacity 0.1s ease-out 0.3s, width 0.1s ease-out 0.5s,
    height 0.2s ease-out 0.3s;
}
</style>
