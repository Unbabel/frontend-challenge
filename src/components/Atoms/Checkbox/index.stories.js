// https://github.com/diegohaz/arc/wiki/Example-components#icon
import React from 'react'
import { storiesOf } from '@storybook/react'
import Checkbox from '.'

storiesOf('Atom/Checkbox', module)
  .add('Default', () => (
    <Checkbox />
  ))
