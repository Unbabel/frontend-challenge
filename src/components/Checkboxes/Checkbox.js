import * as React from 'react'
import '../../resources/styles/checkbox.scss'

const Checkbox = ({ onClick,activeCheckbox }) => {
    return (<React.Fragment>
        <div className="checkbox" >
            <label
                onClick={onClick}
                className={`label-checkbox ${activeCheckbox ? "checked" : ""}`}
            />
        </div>
    </React.Fragment>
    )
}
export default Checkbox;