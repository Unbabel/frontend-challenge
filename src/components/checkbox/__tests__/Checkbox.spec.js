import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";

// Components
import Checkbox from "../Checkbox.vue";

describe("Checkbox", () => {
	let wrapper;

	beforeEach(() => {
		// Arrange
		wrapper = mount(Checkbox, { propsData: { id: "test-checkbox" } });
	});

	it("matches snapshot", () => {
		// Assert
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("is checked by default", () => {
		// Act
		const input = wrapper.find("input");

		// Assert
		expect(input.attributes().checked).toBe("checked");
	});

	it("has correct css classes", () => {
		// Act
		const input = wrapper.find("input");

		// Assert
		expect(wrapper.classes("checkbox-div")).toBe(true);
		expect(input.classes("css-checkbox")).toBe(true);
	});
});
