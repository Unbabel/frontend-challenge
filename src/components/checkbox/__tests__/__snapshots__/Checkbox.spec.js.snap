// Vitest Snapshot v1

exports[`Checkbox > matches snapshot 1`] = `"<div data-v-f2e07ba2=\\"\\" class=\\"checkbox-div\\"><input data-v-f2e07ba2=\\"\\" name=\\"cssCheckbox\\" id=\\"checkbox-test-checkbox\\" type=\\"checkbox\\" aria-checked=\\"\\" checked=\\"checked\\" class=\\"css-checkbox\\"><label data-v-f2e07ba2=\\"\\" for=\\"checkbox-test-checkbox\\"></label></div>"`;
